### PR TITLE
[WFLY-18955] micrometer Galleon layer should contain rules to be discovered

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-18955

- Add rules to layer-spec.xml

### CODE REVIEW

File: galleon-pack/galleon-shared/src/main/resources/layers/standalone/micrometer/layer-spec.xml [hunk 1]
```diff
@@ -8,6 +8,8 @@
         <prop name="org.wildfly.rule.add-on-depends-on" value="only:cdi"/>
         <prop name="org.wildfly.rule.add-on" value="observability,micrometer"/>
```
Comment (jfdenise) [inline galleon-pack/galleon-shared/src/main/resources/layers/standalone/micrometer/layer-spec.xml:9]: @jasondlee , so this layer is also an add-on? Is there a use case to explicitly enable this layer, even if the deployment doesn't use the micrometer API?

Comment (jasondlee) [reply galleon-pack/galleon-shared/src/main/resources/layers/standalone/micrometer/layer-spec.xml:9]: @jfdenise I think so. An admin may want server metrics exported via Micrometer even if no deployments are explicitly using Micrometer.

Comment (jfdenise) [reply galleon-pack/galleon-shared/src/main/resources/layers/standalone/micrometer/layer-spec.xml:9]: @jasondlee , Thank-you!

**Review by jfdenise — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @jasondlee and @jfdenise

**Pull request merged**